src/sourcemap/CodeCoverage.zig

```diff
@@ -687,174 +709,6 @@ pub const ByteRangeMapping = struct {
     }
 };
 
-const CommonCodeCoverageReport = struct {
-    source_url: bun.JSC.ZigString.Slice,
-    executable_lines: Bitset,
-    lines_which_have_executed: Bitset,
-    lines_hits: LinesHits,
-    functions: std.ArrayListUnmanaged(Block),
-    functions_which_have_executed: Bitset,
-    stmts_which_have_executed: Bitset,
-    stmts: std.ArrayListUnmanaged(Block),
-    total_lines: u32 = 0,
-
-    pub const Block = struct {
-        start_line: u32 = 0,
-        end_line: u32 = 0,
-    };
-};
-
-// Coverage report for LCOV format
-pub const LcovCodeCoverageReport = struct {
-    source_url: bun.JSC.ZigString.Slice,
-    executable_lines: Bitset,
-    lines_which_have_executed: Bitset,
-    lines_hits: LinesHits,
-    functions: std.ArrayListUnmanaged(CommonCodeCoverageReport.Block),
-    functions_which_have_executed: Bitset,
-    stmts_which_have_executed: Bitset,
-    stmts: std.ArrayListUnmanaged(CommonCodeCoverageReport.Block),
-    total_lines: u32 = 0,
-
-    pub fn writeFormat(
-        report: *const LcovCodeCoverageReport,
-        base_path: []const u8,
-        writer: anytype,
-    ) !void {
-        var filename = report.source_url.slice();
-        if (base_path.len > 0) {
-            filename = bun.path.relative(base_path, filename);
-        }
-
-        // TN: test name
-        // Empty value appears fine. For example, `TN:`.
-        try writer.writeAll("TN:\n");
-
-        // SF: Source File path
-        // For example, `SF:path/to/source.ts`
-        try writer.print("SF:{s}\n", .{filename});
-
-        // ** Per-function coverage not supported yet, since JSC does not support function names yet. **
-        // FN: line number,function name
-
-        // FNF: functions found
-        try writer.print("FNF:{d}\n", .{report.functions.items.len});
-
-        // FNH: functions hit
-        try writer.print("FNH:{d}\n", .{report.functions_which_have_executed.count()});
-
-        var executable_lines_that_have_been_executed = report.lines_which_have_executed.clone(bun.default_allocator) catch bun.outOfMemory();
-        defer executable_lines_that_have_been_executed.deinit(bun.default_allocator);
-        // This sets statements in executed scopes
-        executable_lines_that_have_been_executed.setIntersection(report.executable_lines);
-        var iter = executable_lines_that_have_been_executed.iterator(.{});
-
-        // ** Branch coverage not supported yet, since JSC does not support those yet. ** //
-        // BRDA: line, block, (expressions,count)+
-        // BRF: branches found
-        // BRH: branches hit
-
-        while (iter.next()) |line| {
-            // DA: line number, hit count
-            // FIXME: Remove the magic number that is used to convert the hit count to the correct value
-            try writer.print("DA:{d},{d}\n", .{ line + 1, report.lines_hits.ptr[line] - 2863311530 });
```

Comment:
@exoego the reason this magic number made it seemingly work: undefined memory. In safety-checked builds of  Zig (debug, ReleaseSafe), `0xa` is the default-initialized value for memory.

This magic number was turning `0xa` into 0. In a release build, this is undefined behavior and will produce a result which will either crash the program, infinite loop, or something unexpected. This is why it did not work in CI.